File: src/main/scala/top/scaleda/verilog/psi/nodes/ReferenceHolder.java
package top.scaleda.verilog.psi.nodes;

import com.intellij.openapi.util.TextRange;
import com.intellij.psi.PsiElement;

public interface ReferenceHolder<T extends PsiElement> {
    T getHoldPsiNode();
    TextRange getHoldPsiNodeRelativeTextRange();
}

File: src/main/scala/top/scaleda/bluespec/psi/nodes/ReferenceHolder.scala
package top.scaleda.bluespec.psi.nodes

import com.intellij.openapi.util.TextRange
import com.intellij.psi.PsiElement

trait ReferenceHolder[T <: PsiElement] {
  def getHoldPsiNode: T
  def getHoldPsiNodeRelativeTextRange: TextRange
}
